File: VSC/C#_ORMs/e_commerce/Models/ProductModel.cs
using System;
using System.ComponentModel.DataAnnotations;
using Stripe;

namespace e_commerce.Models 
{
    public class ProductModel
    {
        [Required]
        [MinLength(2, ErrorMessage = "Product Name must be at least 2 characters long")]
        public string Name{get;set;}
        // public Customer Customer{get;set;}
        public string ImageUrl{get;set;}
        public string Description{get;set;}
        [Required]
        [GreaterThanOne]
        public int Quantity{get;set;}
    }

    public class GreaterThanOneAttribute : ValidationAttribute
    {
        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
        {
            if ((int)value <= 0)
                return new ValidationResult("Products must be greater than zero!");
            return ValidationResult.Success;
        }
    }
}

File: VSC/MERN/FullStackMERN/Product_Manager_2/server/models/product.model.js
const mongoose = require('mongoose')

const ProductSchema = new mongoose.Schema({
    title: {
        type: String,
        minlength: [6, "Name of product must be at least 6 characters"],
        required: [true, "Name of product is required"]
    },
    price: {
        type:Number,
        required: [true, "Price is required"]
    },
    description: {
        type: String,
        minlength: [6, "description of product must be at least 6 characters"],
        required: [true, "description of product is required"]
    }
},{timestamps:true});

module.exports.Product = mongoose.model("Product",ProductSchema);